ndsl/comm/communicator.py

```diff
@@ -93,6 +94,27 @@ def _device_synchronize():
         # this is a method so we can profile it separately from other device syncs
         device_synchronize()
 
+    def _create_all_reduce_quantity(
+        self, input_metadata: QuantityMetadata, input_data
+    ) -> Quantity:
+        """Create a Quantity for all_reduce data and metadata"""
+        all_reduce_quantity = Quantity(
+            input_data,
+            dims=input_metadata.dims,
+            units=input_metadata.units,
+            origin=tuple([0 for dim in input_metadata.dims]),
+            gt4py_backend=input_metadata.gt4py_backend,
+            allow_mismatch_float_precision=True,
```

Comment:
Probably `allow_mismatch_float_precision=False` every quantity should have the same floating point precision

Reply:
Done.